Sources/TensorFlow/Operators/Comparison.swift

```diff
@@ -227,3 +227,16 @@ public extension StringTensor {
         return Raw.equal(self, other)
     }
 }
+
+public extension Tensor where Scalar: TensorFlowFloatingPoint {
+    /// Returns `true` if tensors are of equal shape and all pairs of scalars are approximately
+    /// equal.
+    @inlinable
+    func isAlmostEqual(
+        to other: Tensor,
+        tolerance: Scalar = Scalar.ulpOfOne.squareRoot()
+    ) -> Bool {
+        return self.shape == other.shape &&
```

Comment:
I'm wondering if we want to have this check for mismatching shapes here. @rxwei wouldn't a precondition be better? I believe it would be misleading for this to return `false` when the shapes are mismatching as it should be expected that it operates over tensors with matching shapes.

Another issue is that of broadcasting. If we want to support broadcasting for this function we should remove that check altogether.

Reply:
+1